qiskit/pulse/instructions/frequency.py

```diff
@@ -13,120 +13,111 @@
 """Frequency instructions module. These instructions allow the user to manipulate
 the frequency of a channel.
 """
-from typing import Optional, Union, Tuple
+from typing import Optional, Union
 
 from qiskit.circuit.parameterexpression import ParameterExpression
 from qiskit.pulse.channels import PulseChannel
-from qiskit.pulse.instructions.instruction import Instruction
-from qiskit.pulse.exceptions import PulseError
+from qiskit.pulse.instructions.instruction import FrameInstruction
+from qiskit.pulse.model import Frame, MixedFrame, PulseTarget
 
 
-class SetFrequency(Instruction):
-    r"""Set the channel frequency. This instruction operates on ``PulseChannel`` s.
-    A ``PulseChannel`` creates pulses of the form
+class SetFrequency(FrameInstruction):
+    r"""The set frequency instruction sets the modulation frequency of proceeding pulses
+    associated with the frame the instruction is acting upon.
+
+    In particular, played pulses take the form:
 
     .. math::
         Re[\exp(i 2\pi f jdt + \phi) d_j].
 
-    Here, :math:`f` is the frequency of the channel. The instruction ``SetFrequency`` allows
-    the user to set the value of :math:`f`. All pulses that are played on a channel
-    after SetFrequency has been called will have the corresponding frequency.
+    Here, :math:`f` is the frequency of the modulation. The instruction ``SetFrequency`` allows
+    the user to set the value of :math:`f`.
 
-    The duration of SetFrequency is 0.
+    The duration of ``SetFrequency`` is 0.
     """
 
     def __init__(
         self,
         frequency: Union[float, ParameterExpression],
-        channel: PulseChannel,
+        *,
+        frame: Frame = None,
```

Comment:
```suggestion
        frame: Frame | None = None,
```
and import `from __future__ import annotations`